bound/kt/src/test/kotlin/web5/sdk/vc/VerifiableCredentialTest.kt

```diff
@@ -0,0 +1,286 @@
+    }
+
+    @Test
+    fun test_default_type_added_if_not_supplied() {
+        this.testSuite.include()
+        val vc = VerifiableCredential.create(ISSUER, CREDENTIAL_SUBJECT, VerifiableCredentialCreateOptions())
+        assertEquals(listOf("VerifiableCredential"), vc.type)
+    }
+
+    @Test
+    fun test_default_type_not_duplicated_if_supplied() {
+        this.testSuite.include()
+        val options = VerifiableCredentialCreateOptions(
+            type = listOf("VerifiableCredential")
+        )
+
+        val vc = VerifiableCredential.create(ISSUER, CREDENTIAL_SUBJECT, options)
+        assertEquals(listOf("VerifiableCredential"), vc.type)
+    }
+
+    @Test
+    fun test_developer_provided_type_appended_to_default() {
+        this.testSuite.include()
+        val customType = "CustomType"
+        val options = VerifiableCredentialCreateOptions(
+            type = listOf(customType)
+        )
+
+        val vc = VerifiableCredential.create(ISSUER, CREDENTIAL_SUBJECT, options)
+        assertEquals(listOf("VerifiableCredential", customType), vc.type)
+    }
+
+    @Test
+    fun test_id_generated_if_not_supplied() {
+        this.testSuite.include()
+        val vc = VerifiableCredential.create(ISSUER, CREDENTIAL_SUBJECT, VerifiableCredentialCreateOptions())
+        val uuidPattern = Pattern.compile("^urn:uuid:[0-9a-fA-F-]{36}$")
+        assertTrue(uuidPattern.matcher(vc.id).matches())
+    }
+
+    @Test
+    fun test_id_must_be_set_if_supplied() {
+        this.testSuite.include()
+        val customId = "custom-id"
+        val options = VerifiableCredentialCreateOptions(
+            id = customId
+        )
+
+        val vc = VerifiableCredential.create(ISSUER, CREDENTIAL_SUBJECT, options)
+        assertEquals(customId, vc.id)
+    }
+
+    @Test
+    fun test_issuer_string_must_not_be_empty() {
+        this.testSuite.include()
+        val emptyIssuer = Issuer.StringIssuer("")
+        try {
+            VerifiableCredential.create(emptyIssuer, CREDENTIAL_SUBJECT, VerifiableCredentialCreateOptions())
+            fail("Expected an exception to be thrown")
+        } catch (e: Web5Exception) {
+            when (e) {
+                is Web5Exception.Exception -> {
+                    assertEquals(e.msg, "parameter error issuer id must not be empty")
+                }
+                else -> {
+                    fail("Caught an unexpected exception type")
+                }
+            }
+        }
+    }
+
+    @Test
+    fun test_issuer_string_must_be_set() {
+        this.testSuite.include()
+        val vc = VerifiableCredential.create(ISSUER, CREDENTIAL_SUBJECT, VerifiableCredentialCreateOptions())
+        assertEquals(ISSUER, vc.issuer)
+    }
+
+    @Test
+    fun test_issuer_object_id_must_not_be_empty() {
+        this.testSuite.include()
+        val issuer = Issuer.ObjectIssuer("", "Example Name")
+
+        try {
```

Comment:
if `assertFailsWith` works as the other one can update here too

Reply:
yup done, it's `assertThrows` btw